check_nodes.go

```diff
@@ -18,13 +18,16 @@ package hasql
 
 import (
 	"context"
+	"fmt"
 	"sort"
 	"sync"
 	"time"
 )
 
 type checkedNode struct {
-	Node    Node
+	Node Node
+
+	Primary bool
```

Comment:
I really don't like adding more transitory values to node. This provokes accessing them after check itself when these values might be invalid.

Can you please add a comment before these values making it clear they are transitory and should not be used after the check itself? Also making them private seems like a good idea.